mcp-aws-ec2-server/src/mcp_aws_ec2_server/main.py

```diff
@@ -0,0 +1,241 @@
+                    "state": instance["State"]["Name"],
+                }
+                instances.append(instance_info)
+
+    except Exception as e:
+        return {"error": f"Failed to list instances: {str(e)}"}
+    return {"instances": instances}
+
+
+@mcp.tool()
+async def start_instance(
+    instance_id: Annotated[str, "The ID of the instance to start"],
+    aws_region: Annotated[str, "The AWS region to start the instance in"] = "us-east-1",
+):
+    ec2_client = boto3.client("ec2", region_name=aws_region)
+    try:
+        ec2_client.start_instances(InstanceIds=[instance_id])
+    except Exception as e:
+        return {"error": f"Failed to start instance: {str(e)}"}
+    return {"message": f"Instance {instance_id} started"}
+
+
+@mcp.tool()
+async def stop_instance(
+    instance_id: Annotated[str, "The ID of the instance to stop"],
+    aws_region: Annotated[str, "The AWS region to stop the instance in"] = "us-east-1",
+):
+    ec2_client = boto3.client("ec2", region_name=aws_region)
+    try:
+        ec2_client.stop_instances(InstanceIds=[instance_id])
+    except Exception as e:
+        return {"error": f"Failed to stop instance: {str(e)}"}
+    return {"message": f"Instance {instance_id} stopped"}
+
+
+@mcp.tool()
+async def terminate_instance(
+    instance_id: Annotated[str, "The ID of the instance to terminate"],
+    aws_region: Annotated[
+        str, "The AWS region to terminate the instance in"
+    ] = "us-east-1",
+):
+    ec2_client = boto3.client("ec2", region_name=aws_region)
+    ec2_client.terminate_instances(InstanceIds=[instance_id])
+    return {"message": f"Instance {instance_id} terminated"}
+
+
+@mcp.tool()
+async def reboot_instance(
+    instance_id: Annotated[str, "The ID of the instance to reboot"],
+    aws_region: Annotated[
+        str, "The AWS region to reboot the instance in"
+    ] = "us-east-1",
+):
+    ec2_client = boto3.client("ec2", region_name=aws_region)
+    try:
+        ec2_client.reboot_instances(InstanceIds=[instance_id])
+    except Exception as e:
+        return {"error": f"Failed to reboot instance: {str(e)}"}
+    return {"message": f"Instance {instance_id} rebooted"}
+
+
+@mcp.tool()
+async def create_instance(
+    instance_type: Annotated[str, "The type of instance to create"],
+    image_id: Annotated[str, "The ID of the image to use"],
+    subnet_id: Annotated[str, "The ID of the subnet to use"],
+    key_name: Annotated[str | None, "The name of the key pair to use"] = None,
+    instance_name: Annotated[str | None, "The name of the instance"] = None,
+    tags: Annotated[dict | None, "The tags to apply to the instance"] = None,
+    user_data: Annotated[str | None, "The user data to pass to the instance"] = None,
+    security_group_ids: Annotated[
+        list[str], "The IDs of the security groups to use"
+    ] = [],
+    aws_region: Annotated[
+        str, "The AWS region to create the instance in"
+    ] = "us-east-1",
```

Comment:
```suggestion
    aws_region: Annotated[
        str, "The AWS region to create the instance in, defaults to us-east-1"
    ] = "us-east-1",
```

Reply:
(this should be changed in the other places where we have an `aws_region` parameter too)